### PR TITLE
NEW: Add libgloo

### PR DESCRIPTION
**Pull request by carterbox**
One of the third_party dependecies of pytorch.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

https://github.com/conda-forge/pytorch-cpu-feedstock/issues/108

### CODE REVIEW

**Comment by carterbox** [comment]
Please ignore the linter. It wasn't designed to handle pinnings with build strings.

File: recipes/gloo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# To generate the version number
+{%- if false %}
+git clone git@github.com:facebookincubator/gloo.git
+cd gloo
+git pull
+echo 0.5.0.$(date  +%Y%m%d -d "$(git show -s --format=%ci HEAD)").$(git rev-list HEAD --count).$(git rev-parse --short HEAD)
```
Comment (hmaarrfk) [inline recipes/gloo/meta.yaml:6]: how do you choose 0.5.0?

**Comment by hmaarrfk** [comment]
is the goal to help pytorch build better? if so I would really like an example that shows the resulting simplifications. I never convinced myself that they actually "simplified" the process.

File: recipes/gloo/meta.yaml (hunk 1 above)
Comment (carterbox) [reply recipes/gloo/meta.yaml:6]: This is the version stated in the CMake files! https://github.com/facebookincubator/gloo/blob/main/CMakeLists.txt

**Comment by carterbox** [comment]
The goal is to reduce pytorch build times, not necessarily make it simpler. If your're no longer persuing de-static linking of pytorch dependencies could you please close all of the issues and PRs related to that?

**Comment by hmaarrfk** [comment]
> The goal is to reduce pytorch build times,

The challenge is that I don't really think "gloo" on its own gets us under the 6 hour limit.

if you see any related PRs of mine, it is fine to close them, I'll close them as I come across them.

The challenge was that I needed to:

1. build up tooling to build the exact versions needed by pytorch.
2. build up tooling to adjust the pytorch build process to use pre-compiled libraries.

nothing I did got under the 6 hour limit.

**Comment by carterbox** [comment]
OK. I might try one other thing after re-reading the existing discussions.

**Comment by hmaarrfk** [comment]
don't give up, but I went as far as https://anaconda.org/mark.harfouche/repo#deletePackages

i had "gloo" working, but i had to hack around to get pytorch to detect it , and thats where I stopped.

**Pull request closed without merging**